### PR TITLE
fv/xdp: separate tcp and udp test for clarity

### PR DESCRIPTION
**Pull request by tomastigera**
When things break, it is easier to diagnose it this way.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/fv/xdp_test.go [hunk 1]
```diff
@@ -208,11 +198,11 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ XDP tests with initialized
 	}
 
 	xdpProgramAttached_felix1_eth0 := func() bool {
```
Comment (sridhartigera) [inline felix/fv/xdp_test.go:200]: Can we just rename this to something like xdpProgramAttached_server_eth0.

File: felix/fv/xdp_test.go [hunk 2]
```diff
@@ -208,11 +198,11 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ XDP tests with initialized
...
-		return xdpProgramAttached(felixes[1], "eth0")
+		return xdpProgramAttached(felixes[srvr], "eth0")
 	}
 
 	xdpProgramID_felix1_eth0 := func() int {
```
Comment (sridhartigera) [inline felix/fv/xdp_test.go:204]: Same as above

File: felix/fv/xdp_test.go [hunk 3]
```diff
@@ -99,13 +110,15 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ XDP tests with initialized
 			Expect(err).NotTo(HaveOccurred())
```
Comment (sridhartigera) [inline felix/fv/xdp_test.go:110]: Better to wait for hostEndpoints to be created.

Comment (tomastigera) [reply felix/fv/xdp_test.go:110]: yes, the synchronization here is not great, I am working on fixing this in a separate PR, thanks for the tip!

**Review by sridhartigera — APPROVED** [review]
LGTM

**Pull request merged**